api/src/user/user.handlers.ts

```diff
@@ -78,6 +79,31 @@ async function updateWorksheetCourse(
       ),
     );
 
+  if (worksheetNumber > 0) {
+    const [nameExists] = await db
+      .selectDistinctOn([
+        worksheetNames.netId,
+        worksheetNames.season,
+        worksheetNames.worksheetNumber,
+      ])
+      .from(worksheetNames)
+      .where(
+        and(
+          eq(worksheetNames.netId, netId),
+          eq(worksheetNames.season, season),
+          eq(worksheetNames.worksheetNumber, worksheetNumber),
+        ),
+      );
+    if (!nameExists) {
+      await db.insert(worksheetNames).values({
+        netId,
+        season,
+        worksheetNumber,
+        worksheetName: `Worksheet ${worksheetNumber}`,
+      });
+    }
```

Comment:
I _think_ you can refactor this into a single insert with a conflict-do-nothing, but I'll let @neilsong take look